src/base/http/requestparser.cpp

```diff
@@ -30,314 +31,318 @@
 {
-    // Parse message content
-    qDebug() << Q_FUNC_INFO << "Content-Length: " << m_request.headers["content-length"];
-    qDebug() << Q_FUNC_INFO << "data.size(): " << data.size();
-
-    // Parse url-encoded POST data
-    if (m_request.headers["content-type"].startsWith("application/x-www-form-urlencoded")) {
-        QUrl url;
-        url.setQuery(data);
-        QListIterator<QPair<QString, QString> > i(QUrlQuery(url).queryItems(QUrl::FullyDecoded));
+    // parse POST message-body
+    const QString contentType = m_request.headers[HEADER_CONTENT_TYPE];
+    const QString contentTypeLower = contentType.toLower();
+
+    // application/x-www-form-urlencoded
+    if (contentTypeLower.startsWith(CONTENT_TYPE_FORM_ENCODED)) {
+        QListIterator<QStringPair> i(QUrlQuery(data).queryItems(QUrl::FullyDecoded));
         while (i.hasNext()) {
-            QPair<QString, QString> pair = i.next();
+            const QStringPair pair = i.next();
             m_request.posts[pair.first] = pair.second;
         }
 
         return true;
     }
 
-    // Parse multipart/form data (torrent file)
-    /**
-        data has the following format (if boundary is "cH2ae0GI3KM7GI3Ij5ae0ei4Ij5Ij5")
-
---cH2ae0GI3KM7GI3Ij5ae0ei4Ij5Ij5
-Content-Disposition: form-data; name=\"Filename\"
-
-PB020344.torrent
---cH2ae0GI3KM7GI3Ij5ae0ei4Ij5Ij5
-Content-Disposition: form-data; name=\"torrentfile"; filename=\"PB020344.torrent\"
-Content-Type: application/x-bittorrent
-
-BINARY DATA IS HERE
---cH2ae0GI3KM7GI3Ij5ae0ei4Ij5Ij5
-Content-Disposition: form-data; name=\"Upload\"
-
-Submit Query
---cH2ae0GI3KM7GI3Ij5ae0ei4Ij5Ij5--
-**/
-    QString contentType = m_request.headers["content-type"];
-    if (contentType.startsWith("multipart/form-data")) {
-        const QRegExp boundaryRegexQuoted("boundary=\"([ \\w'()+,-\\./:=\\?]+)\"");
-        const QRegExp boundaryRegexNotQuoted("boundary=([\\w'()+,-\\./:=\\?]+)");
-
-        QByteArray boundary;
-        if (boundaryRegexQuoted.indexIn(contentType) < 0) {
-            if (boundaryRegexNotQuoted.indexIn(contentType) < 0) {
-                qWarning() << "Could not find boundary in multipart/form-data header!";
-                return false;
-            }
-            else {
-                boundary = "--" + boundaryRegexNotQuoted.cap(1).toLatin1();
-            }
+    // multipart/form-data
+    if (contentTypeLower.startsWith(CONTENT_TYPE_FORM_DATA)) {
+        // [rfc2046] 5.1.1. Common Syntax
+
+        // find boundary delimiter
+        const QLatin1String boundaryFieldName("boundary=");
+        const int idx = contentType.indexOf(boundaryFieldName);
+        if (idx < 0) {
+            qWarning() << Q_FUNC_INFO << "Could not find boundary in multipart/form-data header!";
+            return false;
         }
-        else {
-            boundary = "--" + boundaryRegexQuoted.cap(1).toLatin1();
+
+        const QStringRef delimiter = Utils::String::unquote(contentType.midRef(idx + boundaryFieldName.size()));
```

Comment:
You use "delimiter" as QByteArray in every cases. It's better to declare it as QByteArray.

Reply:
ok